FE-service/src/components/subpages/arena/ArenaPage.jsx

```diff
@@ -0,0 +1,136 @@
+import { useState, useEffect } from "react";
+import { CircularProgress, Typography } from "@mui/material";
+import {
+  ArenaContainer,
+  ArenaContent,
+  PokemonListContainer,
+  PokemonListWrapper,
+  PokemonCardContainer,
+  FightButtonContainer,
+  FightButton,
+  EmptyCardImage,
+  TrashIcon,
+  EmptyCardContainer,
+  SummaryContainer,
+} from "./ArenaPage.styles";
+import TrashImage from "../../../icons/trash.png";
+import PokeballImage from "../../../icons/pokeball.png";
+import PokemonCard from "../../shared/pokemonCard/PokemonCard";
+import { useTheme } from "../../../context/ThemeContext";
+import WinnerModal from "./WinnerModal";
+import useHandleFight from "./useHandleFight";
+import { enqueueSnackbar } from "notistack";
+import useGetDbData from "../../../hooks/useGetDbData";
+
+const ArenaPage = () => {
+  const {
+    data: arenaPokemonData,
+    loading: arenaLoading,
+    error: arenaError,
+  } = useGetDbData("arenaPokemons");
+  const {
+    data: arenaFightsData,
+    loading: fightsLoading,
+    error: fightsError,
+  } = useGetDbData("pokemons");
+
+  const [selectedPokemons, setSelectedPokemons] = useState([]);
+  const { theme } = useTheme();
+
+  useEffect(() => {
+    if (arenaPokemonData) {
+      setSelectedPokemons(arenaPokemonData);
+    }
+  }, [arenaPokemonData]);
+
+  const {
+    handleFight,
+    winner,
+    draw,
+    showWinnerModal,
+    handleCloseWinnerModal,
+    handleLeaveArena,
+    removePokemonFromArena,
+  } = useHandleFight(selectedPokemons, arenaFightsData, setSelectedPokemons);
+
+  if (arenaLoading || fightsLoading) {
+    return <CircularProgress />;
+  }
+
+  if (arenaError || fightsError) {
+    enqueueSnackbar(`Error: ${arenaError || fightsError}`, {
+      variant: "error",
+    });
+    return null;
+  }
+
+  return (
+    <ArenaContainer theme={theme}>
+      <Typography variant="h6">
+        The Arena is a location that allows you to perform battle of selected
+        Pokémons.
+      </Typography>
+      {selectedPokemons.length !== 2 && (
+        <Typography variant="h5">Select two Pokémons for fight!</Typography>
+      )}
+      <ArenaContent>
+        <PokemonListContainer>
+          <PokemonListWrapper>
+            <PokemonCardContainer>
+              {selectedPokemons[0] ? (
```

Comment:
The selected Pokémon section should be a separate component because you are currently copying the entire block of code.